### DISCUSSION TITLE
Which embedding model does Vanna uses to convert text to embedding?

### DISCUSSION THREAD

**Discussion by off6atomic** [vanna-ai/vanna#378] [Q&A] [answered]
Is it OpenAI text embedding or some kind of Bert transformer, etc?
Best if you can point to line of code that refers to the model.

Thanks!

**Comment by zainhoda** [comment, marked as answer]
This depends on the setup you're using. Generally speaking for each vector store, it'll be whatever the "default" is. For example, for ChromaDB, it used the default embedding function as defined here:

https://github.com/vanna-ai/vanna/blob/main/src/vanna/chromadb/chromadb_vector.py#L12

Which can then be overridden as necessary.

If you're setting up with vector storage that doesn't include it's own embedding functionality, then you can reference the `generate_embedding` function, which would have an implementation like this (example for using OpenAI):
https://github.com/vanna-ai/vanna/blob/main/src/vanna/openai/openai_embeddings.py#L34-L46
